vsphere/resource_vsphere_host.go

```diff
@@ -0,0 +1,683 @@
+package vsphere
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/terraform-providers/terraform-provider-vsphere/vsphere/internal/helper/hostsystem"
+	"github.com/vmware/govmomi/license"
+
+	"github.com/terraform-providers/terraform-provider-vsphere/vsphere/internal/helper/clustercomputeresource"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/methods"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceVsphereHost() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVsphereHostCreate,
+		Read:   resourceVsphereHostRead,
+		Update: resourceVsphereHostUpdate,
+		Delete: resourceVsphereHostDelete,
+		// Importer: ,
+		Schema: map[string]*schema.Schema{
+			"cluster": {
+				Type:        schema.TypeString,
+				Required:    true,
```

Comment:
Does this need to be required?

Reply:
Yeah, the host is added to a cluster using ClusterComputeResource.AddHost()

Of course now that I'm typing these lines I remembered there's also Folder.AddStandaloneHost() so I guess I can always use that if a cluster is not specified.